frontend/src/components/widgets/DataFrame/columns/utils.ts

```diff
@@ -0,0 +1,358 @@
+ */
+export function toSafeArray(data: any): any[] {
+  if (isNullOrUndefined(data)) {
+    return []
+  }
+
+  if (typeof data === "number" || typeof data === "boolean") {
+    // Single number or boolean
+    return [data]
+  }
+
+  if (typeof data === "string") {
+    if (data === "") {
+      // Empty string
+      return []
+    }
+
+    // Try to parse string to an array
+    if (data.trim().startsWith("[") && data.trim().endsWith("]")) {
+      // Support for JSON arrays: ["foo", 1, null, "test"]
+      try {
+        return JSON.parse(data)
+      } catch (error) {
+        return [data]
+      }
+    } else {
+      // Support for comma-separated values: "foo,1,,test"
+      return data.split(",")
+    }
+  }
+
+  try {
+    const parsedData = JSON.parse(
+      JSON.stringify(data, (_key, value) =>
+        typeof value === "bigint" ? Number(value) : value
+      )
+    )
+    if (!isArray(parsedData)) {
+      return [toSafeString(parsedData)]
+    }
+
+    return parsedData.map((value: any) =>
+      ["string", "number", "boolean", "null"].includes(typeof value)
+        ? value
+        : toSafeString(value)
+    )
+  } catch (error) {
+    return [toSafeString(data)]
+  }
+}
+
+/**
+ * Converts the given value of unknown type to a string without
+ * the risks of any exceptions.
+ *
+ * @param data - The value to convert to a string.
+ *
+ * @return The converted string or a string showing the type of the object as fallback.
+ */
+export function toSafeString(data: any): string {
+  try {
+    try {
+      return toString(data)
+    } catch (error) {
+      return JSON.stringify(data, (_key, value) =>
+        typeof value === "bigint" ? Number(value) : value
+      )
+    }
+  } catch (error) {
+    return `[${typeof data}]`
+  }
+}
+
+/**
+ * Converts the given value of unknown type to a number without
+ * the risks of any exceptions.
+ *
+ * @param value - The value to convert to a number.
+ *
+ * @returns The converted number or null if the value cannot be interpreted as a number.
+ */
+export function toSafeNumber(value: any): number | null {
+  // TODO(lukasmasuch): Should this return null as replacement for NaN?
+
+  if (isNullOrUndefined(value)) {
+    return null
+  }
+
+  if (Array.isArray(value)) {
+    return NaN
+  }
+
+  if (typeof value === "string") {
+    if (value.trim().length === 0) {
+      // Empty string should return null
+      return null
+    }
+
+    try {
+      // Try to convert string to number via numbro:
+      // https://numbrojs.com/old-format.html#unformat
+      const unformattedValue = numbro.unformat(value.trim())
+      if (notNullOrUndefined(unformattedValue)) {
+        return unformattedValue
+      }
+    } catch (error) {
+      // Do nothing here
+    }
+  } else if (value instanceof Int32Array) {
+    // int values need to be extracted this way:
+    // eslint-disable-next-line prefer-destructuring
+    return Number(value[0])
+  }
+
+  return Number(value)
```

Comment:
I think we should cast it as a Number, check if it's NaN and then if it's not return that but it looks like you already have that as a TODO so not sure if this is a conscious choice / would require a bit of refactoring.

return isNaN(Number(value)) ? Number(value) : null

Reply:
Yep, that's what the TODO above is referring to. Haven't really made a decision on that. Currently, it wouldn't make a difference since we anyways to NaN checks in the columns (e.g., see NumberColumn). But most likely to simplify it would be better to just have either `null` or an actual number. But this is probably another fast-follow refactoring after the release.